Sources/Common/DataModel/PointLocator/index.js

```diff
@@ -0,0 +1,848 @@
+      x[2] >= bounds[4] &&
+      x[2] <= bounds[5]
+    ) {
+      return 0.0;
+    }
+
+    const deltas = [0.0, 0.0, 0.0];
+
+    // dx
+    if (x[0] < bounds[0]) {
+      deltas[0] = bounds[0] - x[0];
+    } else if (x[0] > bounds[1]) {
+      deltas[0] = x[0] - bounds[1];
+    }
+
+    // dy
+    if (x[1] < bounds[2]) {
+      deltas[1] = bounds[2] - x[1];
+    } else if (x[1] > bounds[3]) {
+      deltas[1] = x[1] - bounds[3];
+    }
+
+    // dz
+    if (x[2] < bounds[4]) {
+      deltas[2] = bounds[4] - x[2];
+    } else if (x[2] > bounds[5]) {
+      deltas[2] = x[2] - bounds[5];
+    }
+
+    return vtkMath.dot(deltas, deltas);
+  }
+
+  /**
+   * Calculate the squared distance from point x to the bucket "nei".
+   * @param {*} x  The point coordinates
+   * @param {*} nei The bucket coordinates
+   * @returns
+   */
+  function distance2ToBucket(x, nei) {
+    // Compute bucket bounds
+    const bounds = [
+      nei[0] * model.HX + model.BX,
+      (nei[0] + 1) * model.HX + model.BX,
+      nei[1] * model.HY + model.BY,
+      (nei[1] + 1) * model.HY + model.BY,
+      nei[2] * model.HZ + model.BZ,
+      (nei[2] + 1) * model.HZ + model.BZ,
+    ];
+    return distance2ToBounds(x, bounds);
+  }
+
+  /**
+   * Get the neighboring buckets for a given bucket index.
+   * @param {*} ijk The bucket index
+   * @param {*} ndivs The number of divisions in each dimension
+   * @param {*} level The level of neighbors to retrieve
+   * @returns An array of neighboring bucket indices
+   */
+  function getBucketNeighbors(ijk, ndivs, level) {
+    const buckets = [];
+    if (level === 0) {
+      buckets.push([...ijk]);
+      return buckets;
+    }
+    const minLevel = [];
+    const maxLevel = [];
+    for (let i = 0; i < 3; i++) {
+      const min = ijk[i] - level;
+      const max = ijk[i] + level;
+      minLevel[i] = min > 0 ? min : 0;
+      maxLevel[i] = max < ndivs[i] - 1 ? max : ndivs[i] - 1;
+    }
+    for (let i = minLevel[0]; i <= maxLevel[0]; i++) {
+      for (let j = minLevel[1]; j <= maxLevel[1]; j++) {
+        for (let k = minLevel[2]; k <= maxLevel[2]; k++) {
+          if (
+            i === ijk[0] + level ||
+            i === ijk[0] - level ||
+            j === ijk[1] + level ||
+            j === ijk[1] - level ||
+            k === ijk[2] + level ||
+            k === ijk[2] - level
+          ) {
+            buckets.push([i, j, k]);
+          }
+        }
+      }
+    }
+    return buckets;
+  }
+
+  /**
+   * Calculate the overlapping buckets for a given point and distance.
+   * @param {*} x  The point coordinates
+   * @param {*} ijk  The bucket index
+   * @param {*} dist  The search distance
+   * @param {*} level  The level of detail
+   * @returns  An array of overlapping bucket indices
+   */
+  function getOverlappingBuckets(x, ijk, dist, level) {
+    const buckets = [];
+
+    const xBounds = [x[0], x[0], x[1], x[1], x[2], x[2]];
+    const bbox = vtkBoundingBox.newInstance({ bounds: xBounds });
```

Comment:
you do not need to pass a dictionary, you can simply pass xBounds

Reply:
i don't think so i am not using newInstance 

https://github.com/Kitware/vtk-js/blob/master/Sources/Common/DataModel/BoundingBox/index.js#L768-L771